web/Services/LandlordService.cs

```diff
@@ -79,25 +81,66 @@ await using (var transaction = await _dbContext.Database.BeginTransactionAsync(I
         return ILandlordService.LandlordRegistrationResult.Success;
     }
 
+    public async Task<(ILandlordService.LandlordRegistrationResult result, int id)> RegisterLandlord(
+        LandlordProfileModel createModel)
+    {
+        var dbModel = new LandlordDbModel
+        {
+            Title = createModel.Title,
+            FirstName = createModel.FirstName,
+            LastName = createModel.LastName,
+            CompanyName = createModel.CompanyName,
+            Email = createModel.Email,
+            Phone = createModel.Phone,
+            LandlordStatus = createModel.LandlordStatus,
+            LandlordProvidedCharterStatus = createModel.LandlordProvidedCharterStatus
+        };
+        await using (var transaction = await _dbContext.Database.BeginTransactionAsync(IsolationLevel.Serializable))
+        {
+            // Check there isn't already a Landlord with that email. Nothing depends on this currently, but it would probably mean the landlord is a duplicate
+            // This requires Serializable isolation, otherwise it will not lock any rows, and two racing registrations could create duplicate records
+            if (await _dbContext.Landlords.AnyAsync(l => l.Email == createModel.Email))
+                return (ILandlordService.LandlordRegistrationResult.ErrorLandlordEmailAlreadyRegistered,
+                    await GetLandlordIdIfExistsFromModel(dbModel) ?? 0);

```

Comment:
So this returns the id of the landlord with the same email on an email error? That seems odd and I don't think it's necessary.

I'd return null for the id here (need to change the return type to `int? id`), it's a bit clearer than returning 0.